core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -55,6 +55,7 @@ class ExternalWriteConverterCtx(nodeDesc: NodeDesc, totalNumOfRows: Int) extends
   override def put(colIdx: Int, data: Float) = externalFrameWriter.sendFloat(data)
   override def put(colIdx: Int, data: Double) = externalFrameWriter.sendDouble(data)
   override def put(colIdx: Int, data: java.sql.Timestamp) = externalFrameWriter.sendTimestamp(data)
+  override def put(colIdx: Int, data: java.sql.Date) = externalFrameWriter.sendTimestamp(new java.sql.Timestamp(data.getTime))
```

Comment:
Yup, `sendDate` would be the best, we can comment it properly here and meanwhile add this method on the h2o side